Installation fails with R 4.3.0 and gcc 8.2.0

## 1. Bug description

SKAT fails to compile during installation on my system (R version 4.3.0 and gcc version 8.2.0). I tried installing from CRAN as well as directly from GitHub, but both ways the same error is thrown. Note that I was able to successfully install all of SKAT's dependencies -- just not the SKAT package itself.

From the error message (below), it seems that the issue occurs when including the `Rmath.h` header file in `Binary_Permu_SKAT.cpp`.

## 2. Reproducible example

Below is the full command-line output I get when attempting to install SKAT.

```R
> install.packages("SKAT")
trying URL 'https://mirror.las.iastate.edu/CRAN/src/contrib/SKAT_2.2.5.tar.gz'
Content type 'application/x-gzip' length 1272492 bytes (1.2 MB)
==================================================
downloaded 1.2 MB

* installing *source* package ‘SKAT’ ...
** package ‘SKAT’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
using C++ compiler: ‘g++ (GCC) 8.2.0’
/common/software/install/migrated/gcc/8.2.0/bin/g++ -std=gnu++17 -I"/common/software/install/manual/R/4.3.0-openblas-rebuild/lib64/R/include" -DNDEBUG   -I/common/software/install/migrated/bzip2/1.0.6-gcc7.2.0/include/ -L/common/software/install/migrated/bzip2/1.0.6-gcc7.2.0/lib    -fpic  -g -O2  -c Binary_ComputeExact.cpp -o Binary_ComputeExact.o
/common/software/install/migrated/gcc/8.2.0/bin/g++ -std=gnu++17 -I"/common/software/install/manual/R/4.3.0-openblas-rebuild/lib64/R/include" -DNDEBUG   -I/common/software/install/migrated/bzip2/1.0.6-gcc7.2.0/include/ -L/common/software/install/migrated/bzip2/1.0.6-gcc7.2.0/lib    -fpic  -g -O2  -c Binary_ComputeExactMC.cpp -o Binary_ComputeExactMC.o
/common/software/install/migrated/gcc/8.2.0/bin/g++ -std=gnu++17 -I"/common/software/install/manual/R/4.3.0-openblas-rebuild/lib64/R/include" -DNDEBUG   -I/common/software/install/migrated/bzip2/1.0.6-gcc7.2.0/include/ -L/common/software/install/migrated/bzip2/1.0.6-gcc7.2.0/lib    -fpic  -g -O2  -c Binary_ComputeExactSKATO.cpp -o Binary_ComputeExactSKATO.o
/common/software/install/migrated/gcc/8.2.0/bin/g++ -std=gnu++17 -I"/common/software/install/manual/R/4.3.0-openblas-rebuild/lib64/R/include" -DNDEBUG   -I/common/software/install/migrated/bzip2/1.0.6-gcc7.2.0/include/ -L/common/software/install/migrated/bzip2/1.0.6-gcc7.2.0/lib    -fpic  -g -O2  -c Binary_HyperGeo.cpp -o Binary_HyperGeo.o
/common/software/install/migrated/gcc/8.2.0/bin/g++ -std=gnu++17 -I"/common/software/install/manual/R/4.3.0-openblas-rebuild/lib64/R/include" -DNDEBUG   -I/common/software/install/migrated/bzip2/1.0.6-gcc7.2.0/include/ -L/common/software/install/migrated/bzip2/1.0.6-gcc7.2.0/lib    -fpic  -g -O2  -c Binary_Permu_SKAT.cpp -o Binary_Permu_SKAT.o
In file included from Binary_Permu_SKAT.cpp:4:
/common/software/install/manual/R/4.3.0-openblas-rebuild/lib64/R/include/Rmath.h:210:15: error: ‘std::Rf_beta’ has not been declared
 #define beta  Rf_beta
               ^~~~~~~
make: *** [Binary_Permu_SKAT.o] Error 1
ERROR: compilation failed for package ‘SKAT’
* removing ‘/panfs/jay/groups/20/panwei/malak039/R/x86_64-pc-linux-gnu-library/4.3/SKAT’

The downloaded source packages are in
	‘/tmp/RtmpB0N1Nl/downloaded_packages’
Warning message:
In install.packages("SKAT") :
  installation of package ‘SKAT’ had non-zero exit status
```

## 3. Session info

For reference, the output of `utils::sessionInfo()' is included below.

<details>

```R
> utils::sessionInfo()
R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /common/software/install/migrated/openblas/0.3.5_gcc8.2.0_multiarch/lib/libopenblasp-r0.3.5.so;  LAPACK version 3.8.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/Chicago
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.3.0
```
</details>

Hello, 

Same problem here with R 4.3.1 and gcc 8.5.0 

I tried the cran, the github install and also directly via the tar.gz SKAT_2.2.4.tar.gz, but I have the same error here bellow, anyone has a solution for this ? :)

Installing package into ‘/work/backup/gr-fe/R_4.3.1’
(as ‘lib’ is unspecified)
* installing *source* package ‘SKAT’ ...
** using staged installation
** libs
using C++ compiler: ‘g++ (GCC) 8.5.0 20210514 (Red Hat 8.5.0-4)’
/usr/lib64/ccache/g++ -std=gnu++17 -I"/work/scitas-share/spack-r-gr-fe/opt/spack/linux-rhel8-icelake/gcc-8.5.0/r-4.3.1-53ehxii3jkacsemkrdp7u2xau7vakdky/rlib/R/include" -DNDEBUG   -I/usr/local/include    -fpic  -g -O2  -c Binary_ComputeExact.cpp -o Binary_ComputeExact.o
/usr/lib64/ccache/g++ -std=gnu++17 -I"/work/scitas-share/spack-r-gr-fe/opt/spack/linux-rhel8-icelake/gcc-8.5.0/r-4.3.1-53ehxii3jkacsemkrdp7u2xau7vakdky/rlib/R/include" -DNDEBUG   -I/usr/local/include    -fpic  -g -O2  -c Binary_ComputeExactMC.cpp -o Binary_ComputeExactMC.o
/usr/lib64/ccache/g++ -std=gnu++17 -I"/work/scitas-share/spack-r-gr-fe/opt/spack/linux-rhel8-icelake/gcc-8.5.0/r-4.3.1-53ehxii3jkacsemkrdp7u2xau7vakdky/rlib/R/include" -DNDEBUG   -I/usr/local/include    -fpic  -g -O2  -c Binary_ComputeExactSKATO.cpp -o Binary_ComputeExactSKATO.o
/usr/lib64/ccache/g++ -std=gnu++17 -I"/work/scitas-share/spack-r-gr-fe/opt/spack/linux-rhel8-icelake/gcc-8.5.0/r-4.3.1-53ehxii3jkacsemkrdp7u2xau7vakdky/rlib/R/include" -DNDEBUG   -I/usr/local/include    -fpic  -g -O2  -c Binary_HyperGeo.cpp -o Binary_HyperGeo.o
/usr/lib64/ccache/g++ -std=gnu++17 -I"/work/scitas-share/spack-r-gr-fe/opt/spack/linux-rhel8-icelake/gcc-8.5.0/r-4.3.1-53ehxii3jkacsemkrdp7u2xau7vakdky/rlib/R/include" -DNDEBUG   -I/usr/local/include    -fpic  -g -O2  -c Binary_Permu_SKAT.cpp -o Binary_Permu_SKAT.o
In file included from Binary_Permu_SKAT.cpp:4:
/work/scitas-share/spack-r-gr-fe/opt/spack/linux-rhel8-icelake/gcc-8.5.0/r-4.3.1-53ehxii3jkacsemkrdp7u2xau7vakdky/rlib/R/include/Rmath.h:210:15: error: ‘std::Rf_beta’ has not been declared
 #define beta  Rf_beta
               ^~~~~~~
make: *** [/work/scitas-share/spack-r-gr-fe/opt/spack/linux-rhel8-icelake/gcc-8.5.0/r-4.3.1-53ehxii3jkacsemkrdp7u2xau7vakdky/rlib/R/etc/Makeconf:200: Binary_Permu_SKAT.o] Error 1
ERROR: compilation failed for package ‘SKAT’

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1,8 +1,8 @@
 Package: SKAT
 Type: Package 
 Title: SNP-Set (Sequence) Kernel Association Test
-Version: 2.2.5
-Date: 2023-01-12
+Version: 2.2.6
+Date: 2023-11-15
 Author: Seunggeun (Shawn) Lee and Zhangchen Zhao, with contributions from Larisa Miropolsky and Michael Wu
 Maintainer: Seunggeun (Shawn) Lee <lee7801@snu.ac.kr>  
 Description: Functions for kernel-regression-based association tests including Burden test, SKAT and SKAT-O. These methods aggregate individual SNP score statistics in a SNP set and efficiently compute SNP-set level p-values.

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1,10 +1,10 @@
 Package: SKAT
-Type: Package 
+Type: Package
 Title: SNP-Set (Sequence) Kernel Association Test
-Version: 2.2.6
-Date: 2023-11-15
+Version: 2.3.2
+Date: 2025-08-06
 Author: Seunggeun (Shawn) Lee and Zhangchen Zhao, with contributions from Larisa Miropolsky and Michael Wu
-Maintainer: Seunggeun (Shawn) Lee <lee7801@snu.ac.kr>  
+Maintainer: Seunggeun (Shawn) Lee <lee7801@snu.ac.kr>
 Description: Functions for kernel-regression-based association tests including Burden test, SKAT and SKAT-O. These methods aggregate individual SNP score statistics in a SNP set and efficiently compute SNP-set level p-values.
-License: GPL(>=2)
+License: GPL (>= 2)
 Depends: R (>= 2.13.0), Matrix, SPAtest, RSpectra

--- a/src/Binary_Permu_SKAT.cpp
+++ b/src/Binary_Permu_SKAT.cpp
@@ -1,14 +1,14 @@
-#ifndef _STAND_ALONE_
-
-    #include <R.h>
-    #include <Rmath.h>
-    
-#endif
 #include <math.h>
 #include <stdlib.h>
 #include <string.h>
+
 #include "Binary_global.h"
 #include "Binary_Permu_SKAT.h"
+
+#ifndef _STAND_ALONE_
+    #include <R.h>
+    #include <Rmath.h>
+#endif
 
 /*********************************************************************
 

--- a/src/Binary_resampling.cpp
+++ b/src/Binary_resampling.cpp
@@ -1,12 +1,11 @@
-#ifndef _STAND_ALONE_
-
-    #include <R.h>
-    #include <Rmath.h>
-    
-#endif
 #include <math.h>
 #include <stdlib.h>
 #include <string.h>
+
+#ifndef _STAND_ALONE_
+    #include <R.h>
+    #include <Rmath.h>
+#endif
 
 /*********************************************************************
 

--- a/src/kernel_func.cpp
+++ b/src/kernel_func.cpp
@@ -1,6 +1,6 @@
+#include <math.h>
 #include <R.h>
 #include <Rmath.h>
-#include <math.h>
 /********************************************************
 *
 *	Kernel function ver 0.2

